### ISSUE TITLE
[BUG] Failure in creating Azure node/driver and AKS using keys from China region

### ISSUE THREAD

**Issue by smallteeths** [rancher/rancher#43608]
**Rancher Server Setup**
- Rancher version: v2.7.9
- Installation option (Docker install/Helm Chart): Docker

**Describe the bug**

_Use china Azure Cloud Credential_

 1. Create China Azure Cloud Credential Error
    - The 'env' parameter was not passed to the backend 
    <img width="1492" alt="截屏2023-11-24 上午11 51 49" src="https://github.com/rancher/rancher/assets/27733391/ded1e12f-fc9c-492f-b5ff-6e2ff48b8277">
 2. Create rke2 azure 
    -  Js error
    <img width="940" alt="截屏2023-11-24 上午11 54 49" src="https://github.com/rancher/rancher/assets/27733391/16a1916c-901c-49fe-95a3-f8c09a2c6a2c">
    
 3.  The 'aksCheckCredentials' cannot recognize Azure China's Cloud Credential.
  ```
  "failed to get locations: subscription.SubscriptionsClient#ListLocations: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code=\"SubscriptionNotFound\" Message=\"The subscription 'ebca86b2-b2ae-4eb2-a69c-9ee96034073e' could not be found.\""
  ```

**Expected Result**
Rancher should support other regions within Azure.

**Comment by richard-cox** [comment]
Tracking Info
https://github.com/rancher/rancher/issues/43608 is the core issue
https://github.com/rancher/rancher/pull/43605 contains the fix backend side
https://github.com/rancher/dashboard/pull/10071 contains fix frontend side

**Comment by smallteeths** [comment]
> Tracking Info #43608 is the core issue #43605 contains the fix backend side [rancher/dashboard#10071](https://github.com/rancher/dashboard/pull/10071) contains fix frontend side

Thanks for your response. Let me rephrase the `2` issues encountered by the front end:

1. When creating the `node-driver rke2` using a China-based key, the default region for China is not `DEFAULT_REGION = 'westus'`. Our front end should load the region based on the passed key to fill in the default area. If the above process uses a China-based key and loads data using `westus`, it results in the `aksVMSizesV2` interface returning null. Consequently, this triggers an error in `this.vmSizes.filter`.

2. The environment itself is hardcoded during key creation and should be disabled when creating `rke2`.

<img width="1745" alt="截屏2023-11-27 上午11 00 43" src="https://github.com/rancher/rancher/assets/27733391/b03da65c-294b-4264-b9dc-60804f6e98f5">
